simple-adblock/files/simple-adblock.init

```diff
@@ -367,19 +389,54 @@ $(cat $A_TMP)"
 				serviceStatus="${serviceStatus:-'Whitelist processing error'}"
 			fi
 			output 2 'Formatting merged file '
-			if sed "$f_filter" $B_TMP > $A_TMP; then
+			# this egrep filters out non-ascii characters which make dnsmasq
+			# sad. Some lists have wierd unicode characters for homoglyph
+			# attacks but dnsmasq throws an error when trying to parse those
+			# lines.
+			if sed "$f_filter" $B_TMP | egrep -v '[^a-zA-Z0-9=/.-]' > $A_TMP; then
 				output_ok
 			else
 				output_fail
 				serviceStatus="${serviceStatus:-'Data file formatting error'}"
 			fi
 			output 2 'Creating dnsmasq config '
+
```

Comment:
This piece of code should be redundant with Dirk's optimization code. Please comment if I misunderstand.

Reply:
sorry for referencing the blank line, couldn't select the whole code -- it's for removing duplicates when there are local blocklisted domains with duplicates in the lists.

Reply:
It's not redundant. In my case I am blocking a bunch of TLDs (`win`, `bid`, `mobi`, `download`, etc.) yet I see a lot of junk (`local=/52uo5k3t73ypjije.ldsl8m.bid/`) in the generated dnsmasq config. About 700 with my current set of domains. I'd rather just squish all of `.bid` and shrink the dnsmasq config in the process. That's why I filtered it after Dirk's optimizations.

Reply:
Here's the different that regex filter makes:
```
-rw-r--r--    1 root     root        771744 Nov 27 00:43 simple-adblock.ck
-rw-r--r--    1 root     root        964137 Nov 27 00:02 simple-adblock.old
-rw-r--r--    1 root     root        229333 Nov 27 00:43 simple-adblock.gz.ck
-rw-r--r--    1 root     root        285072 Nov 27 00:02 simple-adblock.gz.old

# wc -l simple-adblock.ck simple-adblock.old 
    28097 simple-adblock.ck
    34411 simple-adblock.old
```

The `.ck` version has my extra regex filtering enabled.

Reply:
I see, I was under impression that Dirk's algorithm would filter those out as well. Maybe you can modify that code rather than implementing another filter?

Reply:
I'll think about that. Dirk's filter OOMs on my GL-USB150 fairly frequently; my grep has not yet triggered any OOMs.

Reply:
I had a look at this and if I'm reading it right, his filter assumes the existence of a `.` in the domain, and multiple elements (`example`, `.`, `tld`). In my case, I'm going after whole TLDs.